.github/workflows/.test.yml

```diff
@@ -0,0 +1,427 @@
+# reusable workflow
+name: .test
+
+# TODO: hide reusable workflow from the UI. Tracked in https://github.com/community/community/discussions/12025
+
+on:
+  workflow_call:
+    inputs:
+      use_snapshotter:
+        required: true
+        type: boolean
```

Comment:
Will calling the workflow twice with different inputs see any conflicts in the testsuite artifacts? I assume GitHub's own artifacts are fine as they're separate jobs, but does for example the codecov upload need to distinguish them? (I didn't see any other artifact uploads after a quick look).

Or should it distinguish them? I guess technically we want total coverage for the whole testsuite run, including both graphdrivers and snapshotters. Although the snapshotter specific coverage will be zero in usual test-runs, so maybe codecov upload should be disabled for `use_snapshotter=true` so we don't get spurious falls in code-coverage on merge-to-main runs, if that's a concern.

I note that Windows has a `send_coverage` input which is only used on Windows 2022, so I assume we want to skip-non-primary test-runs, so my guess is that the best course is to skip the `codecov/codecov-action` steps if `use_snapshotter` is true.

Reply:
I don't see conflicts in the artifacts, the test runs reports are acurate for example.

Gotta check codecov though

Reply:
I think it's fine sending the coverage in both cases, windows has the `send_coverage` because it's testing the same thing on two different windows versions. Whereas with the containerd snapshotter we are testing a completely different code path

Reply:
Reports are no longer broken :)